File: CodeRun_Boost/7264_Bridge_to_the_palace/solution.java
import java.util.Arrays;

class Solution {
    public int solve(int n, int[] a) {
        Arrays.sort(a);

        int left = 0;
        int maxKeep = 0;

        for (int right = 0; right < n; ++right) {
            while (a[right] - a[left] >= n) {
                left++;
            }

            int currentKeep = right - left + 1;
            if (currentKeep > maxKeep) {
                maxKeep = currentKeep;
            }
        }

        int minMoves = n - maxKeep;
        return minMoves;
    }
}

File: CodeRun_Boost/7264_Bridge_to_the_palace/solution.go
package main

import "sort"

func solution(n int, a []int) int {
	sort.Ints(a)

	left := 0
	maxKeep := 0

	for right := 0; right < n; right++ {
		for a[right]-a[left] >= n {
			left++
		}

		currentKeep := right - left + 1
		if currentKeep > maxKeep {
			maxKeep = currentKeep
		}
	}

	minMoves := n - maxKeep
	return minMoves
}